File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
        
    dic = {}
    for i in range(len(tangerine)):
        if tangerine[i] in dic:
            dic[tangerine[i]] += 1
        else:
            dic[tangerine[i]] = 1
    
    sorted_dic = sorted(dic.items(), key=lambda item: item[1], reverse=True)
    
    cnt = 0
    for a, b in sorted_dic:
        k = k - b
        cnt += 1
        if k <= 0:
            break
    
    return cnt

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <bits/stdc++.h>
using namespace std;
int total = 0; 
int answer = 0;

int solution(int k, vector<int> tangerine) {
    int maxNum = *max_element(tangerine.begin(), tangerine.end());
    vector<int> cnt(maxNum+1, 0);
    
    for(int i=0; i<tangerine.size(); i++) {
        cnt[tangerine[i]]++;
    }

    sort(cnt.begin(), cnt.end(), greater<int>());
    
    for(int i=0; i<cnt.size(); i++) {
        total += cnt[i];
        answer++;
        if(total >= k) {
            break;
        }
    }
    
    return answer;
}